tests/rest_api/test_rest_api_source.py

```diff
@@ -26,26 +28,82 @@ def test_rest_api_source(destination_name: str) -> None:
             }
         },
         "resources": [
-            "pokemon",
+            {
+                "name": "pokemon_list",
+                "endpoint": "pokemon",
+            },
             "berry",
             "location",
         ],
     }
-
-    load_info = pipeline.run(rest_api_source(config))
+    data = rest_api_source(config)
+    pipeline = _make_pipeline(destination_name)
+    load_info = pipeline.run(data)
     print(load_info)
     assert_load_info(load_info)
     table_names = [t["name"] for t in pipeline.default_schema.data_tables()]
     table_counts = load_table_counts(pipeline, *table_names)
 
-    assert table_counts.keys() == {"pokemon", "berry", "location"}
+    assert table_counts.keys() == {"pokemon_list", "berry", "location"}
 
-    assert table_counts["pokemon"] == 1302
+    assert table_counts["pokemon_list"] == 1302
     assert table_counts["berry"] == 64
     assert table_counts["location"] == 1036
 
 
 # TODO: Add incorrect config test
 # - incorrect default_resource (missing endpoint, nested params)
 # - incorrect resources
-# - incorrect key (default_resource)
\ No newline at end of file
+# - incorrect key (default_resource)
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_dependent_resource(destination_name: str) -> None:
+    config = {
+        "client": {
+            "base_url": "https://pokeapi.co/api/v2/",
+        },
+        "resource_defaults": {
+            "endpoint": {
+                "params": {
+                    "limit": 1000,
+                },
+            }
+        },
+        "resources": [
+            {
+                "name": "pokemon_list",
+                "endpoint": {
+                    "path": "pokemon",
+                    "params": {
+                        "limit": "2",
```

Comment:
This will result in even more requests, since it's only 2 items per page. Limiting should be done with `add_limit` to the resource, but that'd may not work for transformer resource (the `pokemon`) below, similar to test connection function from another PR. Need to think how to make it work actually since it's a frequent use case: to test the full source load.

Reply:
I agree that many APIs work like that. But the Pokemon API returns indeed only 2 records if you query it with `limit=2`. See my output:

```
curl 'https://pokeapi.co/api/v2/pokemon?limit=2' | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   232    0   232    0     0    212      0 --:--:--  0:00:01 --:--:--   212
{
  "count": 1302,
  "next": "https://pokeapi.co/api/v2/pokemon?offset=2&limit=2",
  "previous": null,
  "results": [
    {
      "name": "bulbasaur",
      "url": "https://pokeapi.co/api/v2/pokemon/1/"
    },
    {
      "name": "ivysaur",
      "url": "https://pokeapi.co/api/v2/pokemon/2/"
    }
  ]
}
```

The challenge I see is that calling `add_limit(2)` on a paginated resource means to stop yielding after two __pages__ of records, not after two records. Thus, it is unpredictable how many records will be contained in two pages because it depends on the request and the database.

**Example 1:** Workable account A1 can have 3 jobs and account A2 has 300 jobs. When the page_size is 100 then `add_limit(2)` will yield 3 jobs for the first account and 200 for the second. Thus, `add_limit(n)` yields `n <= length(records)`

**Example 2:** Consider the same accounts A1 and A2 as above. When the page_size is 100 then `add_limit(1)` will yield 3 jobs for the first account and 100 for the second. Thus, `add_limit(n)` yields `n >= length(records)` for A1 and `n <= length(records)` for A2. I was confused when I discovered that.

Test code often serves as an example of proper usage. Thus, I'd consider it misleading to use `add_limit(n)` in our example code as long as `n != length(records)` because for non-paginated resources `add_limit(n)` yields `n == length(records)`.

What are your thoughts?